Mathlib/Data/List/Basic.lean

```diff
@@ -1231,6 +1231,12 @@ theorem forall_cons (p : α → Prop) (x : α) : ∀ l : List α, Forall p (x ::
   | [] => (and_iff_left_of_imp fun _ ↦ trivial).symm
   | _ :: _ => Iff.rfl
 
+theorem forall_append (p : α → Prop) (xs ys : List α) :
+    Forall p (xs ++ ys) ↔ Forall p xs ∧ Forall p ys := by
+  match xs with
+  | [] => simp
+  | _ :: _ => rw [cons_append, forall_cons, forall_cons, forall_append, and_assoc]
```

Comment:
```suggestion
theorem forall_append (p : α → Prop) : ∀ (xs ys : List α),
    Forall p (xs ++ ys) ↔ Forall p xs ∧ Forall p ys
  | [] => by simp
  | _ :: _ => by simp [forall_append, and_assoc]
```
minigolf. 
Maybe this should also be a simp theorem? cc @YaelDillies

Reply:
Yes, please make it simp and make `p` implicit (variables should usually be implicit in iffs)

Reply:
Thanks for the suggestions! Will update soon

Reply:
I also made `xs` and `ys` following the suggestion and the style of `forall_iff_forall_mem` below. Is that alright?

Reply:
Yep, that's perfect!